app/utils/window-utils.js

```diff
@@ -0,0 +1,37 @@
+"use strict";
+
+let { BrowserWindow } = require("electron");
+
+/**
+ * Collection of Window Utilities
+ *
+ * @class WindowUtils
+ */
+class WindowUtils {
+	/**
+	 * Attempt to open the devtools with
+	 * a fallback if the regular method fails.
+	 *
+	 * @static
+	 * @param {Electron.BrowserWindow} window
+	 * @param {Electron.OpenDevToolsOptions["mode"]} mode
+	 * @memberof WindowUtils
+	 */
+	static openDevtools(window, mode = "right") {
+		window.webContents.openDevTools({ mode });
+
+		// Fallback if openDevTools fails
+		if (!window.webContents.isDevToolsOpened()) {
```

Comment:
This detection seems to be scuffed, it will return `false` even on non bugged environments. If I wait a bit before running this, it returns `true`.